core/router/gateway_status.go

```diff
@@ -21,11 +20,5 @@ func (r *router) HandleGatewayStatus(gatewayID string, status *pb_gateway.Status
 		}
 	}()
 
-	gateway := r.getGateway(gatewayID)
-	gateway.LastSeen = time.Now()
-	err = gateway.Status.Update(status)
-	if err != nil {
-		return err
-	}
-	return nil
+	return r.getGateway(gatewayID).HandleStatus(status)
```

Comment:
We have special error handling in this func:

```go
  	defer func() {
  		if err != nil {
 			ctx.WithError(err).Warn("Could not handle gateway status")
 		} else {
 			ctx.WithField("Duration", time.Now().Sub(start)).Info("Handled gateway status")
  		}
  	}()
```

For this to work, `err` needs to be set